hibernate-rx-core/src/main/java/org/hibernate/rx/collection/RxPersistentList.java

```diff
@@ -0,0 +1,42 @@
+package org.hibernate.rx.collection;
+
+import org.hibernate.HibernateException;
+import org.hibernate.collection.internal.PersistentList;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.loader.CollectionAliases;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.rx.util.impl.RxUtil;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * A {@link PersistentList} that makes sure that {@link CompletionStage} values complete before adding them
+ * to the underlying list;
+ */
+public class RxPersistentList extends PersistentList {
```

Comment:
@DavideD what was the motivation for introducing this collection-related stuff in this work, bearing in mind that we currently document the following limitations:

> At this time, Hibernate RX does _not_ support the following features:
>
> - `@ElementCollection` and `@ManyToMany`
> - `@OneToMany` without `mappedBy` 

I was surprised to see a bunch of persistent collection-related stuff in here, because I didn't think we had tests that would exercise these features.

Reply:
See [RxPersistentList#readFrom](https://github.com/hibernate/hibernate-rx/pull/66/files/299835cf057732e91bc02ba2aabaa8e238b7fbd5#diff-ae92aa6395093fca5730bf179aa0a9ffR26). I needed to resolve the CompletionStage before setting the value in the List. That's the only way I've found to achieve this.

Reply:
Well I hope there's a better way to solve it.

Because note that it's not going to be enough to just introduce `RxPersistentList` without `RxPersistentSet`, `RxPersistentSortedSet`, etc, etc.

Would *really* prefer to not go down that path.